### ISSUE TITLE
rho scan state a success even if a task fails

### ISSUE THREAD

**Issue by elyezer** [quipucords/rho#137]
The controller and the target host are fedora machines, in fact, both are the same machine. The profile is scanning 127.0.0.1.

This is the output of the scan command:

```console
$ rho scan --reset --reportfile report.csv --profile 1 --facts default
Please enter your rho vault password: 

# omitting the tasks that have passed

TASK [redhat_packages : initialize redhat-packages.last_built to dictionary] ***
ok: [127.0.0.1] => (item=redhat-packages.last_built) => {"ansible_facts": {"redhat_packages": {"redhat-packages.is_redhat": "", "redhat-packages.last_built": "", "redhat-packages.last_installed": "", "redhat-packages.num_installed_packages": "", "redhat-packages.num_rh_packages": ""}}, "changed": false, "item": "redhat-packages.last_built"}

TASK [write : write the list to a csv] *****************************************
fatal: [127.0.0.1]: FAILED! => {"changed": false, "failed": true, "module_stderr": "Shared connection to 127.0.0.1 closed.\r\n", "module_stdout": "Traceback (most recent call last):\r\n  File \"/tmp/ansible_Ww5Eik/ansible_module_spit_results.py\", line 21, in <module>\r\n    from past.builtins import long  # pylint: disable=redefined-builtin\r\nImportError: No module named past.builtins\r\n", "msg": "MODULE FAILURE", "rc": 0}
	to retry, use: --limit @/home/elyezer/code/rho/rho_playbook.retry

PLAY RECAP *********************************************************************
127.0.0.1                  : ok=143  changed=72   unreachable=0    failed=1   

Scanning has completed. The mapping has been stored in file '1_host_auth_map'. The facts have been stored in '/home/elyezer/code/rho/report.csv'

$ cat report.csv
cat: report.csv: No such file or directory
```
The error mentions about something wrong with a python module, the environment on the controller machine is:

```
$ pip freeze
alabaster==0.7.10
ansible==2.3.1.0
asn1crypto==0.22.0
astroid==1.5.3
Babel==2.4.0
backports.functools-lru-cache==1.4
bcrypt==3.1.3
certifi==2017.4.17
cffi==1.10.0
chardet==3.0.4
configparser==3.5.0
coverage==4.4.1
cryptography==2.0
docutils==0.13.1
enum34==1.1.6
flake8==3.3.0
funcsigs==1.0.2
future==0.16.0
idna==2.5
imagesize==0.7.1
ipaddress==1.0.18
isort==4.2.15
Jinja2==2.9.6
lazy-object-proxy==1.3.1
MarkupSafe==1.0
mccabe==0.6.1
mock==2.0.0
paramiko==2.2.1
pbr==3.1.1
pexpect==4.2.1
ptyprocess==0.5.2
py==1.4.34
pyasn1==0.2.3
pycodestyle==2.3.1
pycparser==2.18
pycrypto==2.6.1
pyflakes==1.5.0
Pygments==2.2.0
pylint==1.7.2
PyNaCl==1.1.2
pytest==3.1.3
pytest-cov==2.5.1
pytz==2017.2
PyYAML==3.12
requests==2.18.1
-e git+git@github.com:quipucords/rho.git@c586ed34df4f2427dd62f5eb6caeee504bbbcce3#egg=rho
singledispatch==3.4.0.3
six==1.10.0
snowballstemmer==1.2.1
Sphinx==1.6.3
sphinxcontrib-websupport==1.0.1
typing==3.6.1
urllib3==1.21.1
wrapt==1.10.10
```

**Comment by chambridge** [comment]
Strange you hit this import error `pip install future` is suppose to provide the `past` module (https://pypi.python.org/pypi/past/) and from you info above you have `future==0.16.0` which is the same as on my system.

This is separate from the incorrect output message which needs to be fixed. I'm just uncertain about the dependency problem.

**Comment by chambridge** [comment]
@elyezer Any updates on this dependency issue. Any further detail you can provide?

**Comment by elyezer** [comment]
@chambridge I have not tried the scan command in a while, let me check.

**Comment by elyezer** [comment]
Tried to run scan, but I am facing:

```console
$ rho scan --reset --reportfile report.csv --profile 1                
Please enter your rho vault password: 
Traceback (most recent call last):
  File "/home/elyezer/.virtualenvs/camayoc/bin/rho", line 6, in <module>
    exec(compile(open(__file__).read(), __file__, 'exec'))
  File "/home/elyezer/code/rho/bin/rho", line 28, in <module>
    CLI().main()
  File "/home/elyezer/code/rho/rho/cli.py", line 124, in main
    cmd.main()
  File "/home/elyezer/code/rho/rho/clicommand.py", line 91, in main
    self._do_command()
  File "/home/elyezer/code/rho/rho/scancommand.py", line 404, in _do_command
    self.verbosity)
  File "/home/elyezer/code/rho/rho/scancommand.py", line 58, in _create_ping_inventory
    hostname = str_to_ascii(profile_range)
  File "/home/elyezer/code/rho/rho/utilities.py", line 207, in str_to_ascii
    return in_string.decode('utf-8').encode('ascii')
AttributeError: 'str' object has no attribute 'decode'
```

I am running RHO on python:

```console
$ python --version
Python 3.6.2
```

**Comment by elyezer** [comment]
Tested on python 2.7.13 and the dependency issue continues:

```console
$ rho scan --reset --reportfile report.csv --profile 

...

TASK [write : write the list to a csv] *****************************************
ythonfatal: [127.0.0.1]: FAILED! => {"changed": false, "failed": true, "module_stderr": "Shared connection to 127.0.0.1 closed.\r\n", "module_stdout": "Traceback (most recent call last):\r\n  File \"/tmp/ansible_hEwepZ/ansible_module_spit_results.py\", line 21, in <module>\r\n    from past.builtins import long  # pylint: disable=redefined-builtin\r\nImportError: No module named past.builtins\r\n", "msg": "MODULE FAILURE", "rc": 0}
	to retry, use: --limit @/home/elyezer/code/rho/rho_playbook.retry

PLAY RECAP *********************************************************************
127.0.0.1                  : ok=156  changed=70   unreachable=0    failed=1   

An error has occurred during the scan. Please review the output to resolve the given issue.

$ python --version
Python 2.7.13

```

**Comment by chambridge** [comment]
I pulled a Python 2.7.13 docker container, performed an `pip install -r requirements.txt`, created an auth and profile, and performed the scan successfully. Not sure how to reproduce.

Are you running this in a virtualenv?
Can you double check future with `pip install --upgrade future`?

**Comment by chambridge** [comment]
FYI - @elyezer 
I made fixed for #97 and performed a scan using a Python 3.6 image.

**Comment by elyezer** [comment]
I've investigate this issue a little bit more and here are my findings. I have a Python 3.6 virtualenv which is running RHO installed from git and I am trying to scan the localhost system which is Fedora 26.

So what is happening is the requirements are installed properly on the controller machine (the Python 3.6 venv). But when ssh into the target (same machine) it is running on Python 2 and there is no python-future installed in there. That said, the issue is that python-future needs to be installed on the target machine, otherwise that import error will be raised.

That requirement is needed split results and is not on the standard library, I think we have two approaches 1) inform the user that each target machine will need python-future available 2) remove that requirement from the split_results. I like the latter most because it does not require any extra effort from the user (imagine they installing that package on many machines) and it is very simple to fix the issue without requiring a library, it would be something like this:

```python
>>> import sys
>>> if sys.version_info > (3,):
...     long = int
>>> long(1)
1L
```

Specially because the module split_results.py already has a if block checking for py3.
